src/components/App/App.js

```diff
@@ -53,13 +57,31 @@ class App extends Component {
 }
 
   render() {
+
+    let mainView;
+    if(!this.state.collectionView){
+      mainView = <MovieInfoView movie={this.state.currentMovie} handleMovieView={this.handleMovieView}/>
+    } else if (this.state.movies.length > 0) {
```

Comment:
I like this refactor to keep the return clean. It was interesting to learn too about the async problems that setState sometimes causes. It makes sense to have the render happen if the length of the movies.